From 0b691502fec476e0524cbed09ebb06a5ae5eeb95 Mon Sep 17 00:00:00 2001
From: Maciej Czekaj <mczekaj@marvell.com>
Date: Tue, 12 Nov 2019 12:28:15 +0100
Subject: [PATCH 408/767] arm64: Fix workaround for Marvell erratum 37119

commit 89352f7e4fd4301e7a5f2043502bf5492fd3378c from
git@git.assembla.com:cavium/WindRiver.linux.git

Workaround for erratum 37119 used the
alternative_insn macro in an incorrect way
effectively producing nops instead of tlbi.

Change-Id: I111e5bbcf3ef7007ad1acd63d33d1153ac451bb2
Signed-off-by: Maciej Czekaj <mczekaj@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/18098
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/kernel/entry.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/kernel/entry.S b/arch/arm64/kernel/entry.S
index 4db2bf6a0e84..a4597599dbd7 100644
--- a/arch/arm64/kernel/entry.S
+++ b/arch/arm64/kernel/entry.S
@@ -365,7 +365,7 @@ alternative_else_nop_endif
 	add	sp, sp, #S_FRAME_SIZE		// restore sp
 
 #ifdef CONFIG_MRVL_ERRATUM_37119
-alternative_insn "tlbi aside1, xzr", "nop", ARM64_WORKAROUND_MRVL_37119
+alternative_insn nop, "tlbi aside1, xzr", ARM64_WORKAROUND_MRVL_37119
 #endif
 	.if	\el == 0
 alternative_insn eret, nop, ARM64_UNMAP_KERNEL_AT_EL0
-- 
2.31.1

